proxy.go

```diff
@@ -1,13 +1,33 @@
 package solidproxy
 
 import (
+	"crypto/tls"
 	"io/ioutil"
 	"net/http"
 	"net/url"
 	"time"
 )
 
-func ProxyHandler(w http.ResponseWriter, req *http.Request) {
+const (
+	MaxIdleConnections int = 20
+	RequestTimeout     int = 5
+)
+
+type Proxy struct {
+	HttpClient      *http.Client
+	HttpAgentClient *http.Client
```

Comment:
At first I found having both `HttpClient` and `HttpAgentClient` to be confusing in this struct, especially since there's another `Agent` struct, and at first I assumed the agent client to be part of the agent struct.

It looks like these two client objects are for requesting with/without WebID/TLS credentials... correct? Is there another way to achieve this behavior without storing both client objects in the struct?